### ISSUE TITLE
Python 3.x support

### ISSUE THREAD

**Issue by vwiencek** [hazelcast/hazelcast-python-client#29]
I'm trying to access an Hazelcast map through a python client. I'm trying to execute file [https://github.com/hazelcast/hazelcast-python-client/blob/master/examples/main.py](url)

I get the following error: `ImportError: No module named 'Queue'`

I'm running python 3.5.2. It seems that the python client is not made for python 3 .....

**Comment by asimarslan** [comment]
Hi @vwiencek , Hazelcast Python Client only support 2.7 at the moment. We are planning to support python 3.x but not for the first release. I'll keep this issue open so that we can track python 3.x support development.

**Comment by vwiencek** [comment]
Ok thanks ... looking forward to it

**Comment by bzeyben** [comment]
Hi @vwiencek, Any more updates on this?

**Comment by qickrooms** [comment]
Bump:)

Any timeline on possible python3 support. I am just starting to investigate connecting my existing java hazecast client and first python clients together, don't really want to use python 2 if can help it.

**Comment by waseem18** [comment]
Hi @asimarslan 

I'm interested in working on this issue.

Do you want me to strictly port it to Python 3 or should it support both Python 2 & 3?

Thank You.

**Comment by asimarslan** [comment]
Hi @waseem18,

We cannot drop python 2 support as many users still use it on production so it should support both 2 & 3.

Please note that you should sign [the contributor agreement](https://hazelcast.atlassian.net/wiki/spaces/COM/pages/6357071/Hazelcast+Contributor+Agreement) before we can merge your PR

Thanks for your support.

**Comment by waseem18** [comment]
Thanks for the reply @asimarslan . I'll start working and report the progress here :+1:

**Comment by dbrimley** [comment]
@waseem18 Great community support for this, big thanks to you.  Eager to see what you come up with.

**Comment by waseem18** [comment]
Started working on it today! Will put a Work in progress PR when I'm done with some reasonable part. :)

**Comment by waseem18** [comment]
Update on this issue : I've been working on making `hazelcast-python-client` compatible with both Python 2 and 3 and made the required changes.

I found that this python client uses 'hazelcast-remote-controller' (hzrc which uses thrift 0.10.0) which got some non Python 2 & 3  compatible code. I'm looking into this part now.

I've got three straight holidays now, so will be fully working on this stuff and then will put a PR to both `hzrc` and this python client.

:)
